src/test/java/io/debezium/connector/vitess/VitessValueConverterTest.java

```diff
@@ -117,8 +124,24 @@ public void shouldGetInt64SchemaBuilderForTime() throws InterruptedException {
         assertThat(schemaBuilder.schema().type()).isEqualTo(SchemaBuilder.INT64_SCHEMA.type());
     }
 
+    @Test
+    public void shouldGetStringSchemaBuilderForTemporalTypesWithIsoStringTimePrecisionMode() throws InterruptedException {
+        VitessConnectorConfig config = new VitessConnectorConfig(TestHelper.defaultConfig()
+                .with(VitessConnectorConfig.TIME_PRECISION_MODE.name(), TemporalPrecisionMode.ISOSTRING).build());
+        setUpResources(config);
+        decoder.processMessage(temporalFieldEvent(), null, null, false);
+        Table table = schema.tableFor(TestHelper.defaultTableId());
+
+        for (String columnName : List.of("date_col", "time_col", "datetime_col")) {
+            Column column = table.columnWithName(columnName);
+            SchemaBuilder schemaBuilder = converter.schemaBuilder(column);
+            assertThat(schemaBuilder.schema()).isEqualTo(Schema.STRING_SCHEMA);
+        }
+    }
+
     @Test
     public void shouldGetYearSchemaBuilderForYear() throws InterruptedException {
+        setUpResources(TestHelper.defaultConnectorConfig());
```

Comment:
Maybe instead of adding the explicit calls you could provide a dedicated test class that would test temporal string conversion only?

Reply:
Great idea, updated!